diff --git a/third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc b/third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc
--- a/third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc
+++ b/third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc
@@ -33,6 +33,7 @@
 #include <memory>
 #include <utility>
 
+#include "base/command_line.h"
 #include "base/bit_cast.h"
 #include "base/containers/contains.h"
 #include "base/feature_list.h"
@@ -3855,5 +3856,10 @@
     case WebGLDebugRendererInfo::kUnmaskedRendererWebgl:
       if (ExtensionEnabled(kWebGLDebugRendererInfoName)) {
+        const auto* cosmium_cmd = base::CommandLine::ForCurrentProcess();
+        if (cosmium_cmd->HasSwitch("cosmium-webgl-renderer")) {
+          return WebGLAny(script_state, String::FromUTF8(
+              cosmium_cmd->GetSwitchValueASCII("cosmium-webgl-renderer")));
+        }
         if (IdentifiabilityStudySettings::Get()->ShouldSampleType(
                 blink::IdentifiableSurface::Type::kWebGLParameter)) {
           RecordIdentifiableGLParameterDigest(
@@ -3870,5 +3876,10 @@
     case WebGLDebugRendererInfo::kUnmaskedVendorWebgl:
       if (ExtensionEnabled(kWebGLDebugRendererInfoName)) {
+        const auto* cosmium_cmd = base::CommandLine::ForCurrentProcess();
+        if (cosmium_cmd->HasSwitch("cosmium-webgl-vendor")) {
+          return WebGLAny(script_state, String::FromUTF8(
+              cosmium_cmd->GetSwitchValueASCII("cosmium-webgl-vendor")));
+        }
         if (IdentifiabilityStudySettings::Get()->ShouldSampleType(
                 blink::IdentifiableSurface::Type::kWebGLParameter)) {
           RecordIdentifiableGLParameterDigest(
