diff --git a/third_party/blink/renderer/core/frame/navigator_id.cc b/third_party/blink/renderer/core/frame/navigator_id.cc
index 22243ebf5e..a246f64175 100644
--- a/third_party/blink/renderer/core/frame/navigator_id.cc
+++ b/third_party/blink/renderer/core/frame/navigator_id.cc
@@ -31,6 +31,7 @@
 
 #include "third_party/blink/renderer/core/frame/navigator_id.h"
 
+#include "base/command_line.h"
 #include "base/feature_list.h"
 #include "build/build_config.h"
 #include "third_party/blink/public/common/features.h"
@@ -59,6 +60,11 @@ String NavigatorID::appVersion() {
 }
 
 String NavigatorID::platform() const {
+  const auto* cosmium_cmd = base::CommandLine::ForCurrentProcess();
+  if (cosmium_cmd->HasSwitch("cosmium-platform")) {
+    return String::FromUtf8(
+        cosmium_cmd->GetSwitchValueASCII("cosmium-platform"));
+  }
 #if BUILDFLAG(IS_MAC)
   // Match Safari and Mozilla on Mac x86.
   return "MacIntel";
