diff --git a/third_party/blink/renderer/core/frame/navigator_concurrent_hardware.cc b/third_party/blink/renderer/core/frame/navigator_concurrent_hardware.cc
--- a/third_party/blink/renderer/core/frame/navigator_concurrent_hardware.cc
+++ b/third_party/blink/renderer/core/frame/navigator_concurrent_hardware.cc
@@ -4,12 +4,23 @@
 
 #include "third_party/blink/renderer/core/frame/navigator_concurrent_hardware.h"
 
+#include "base/command_line.h"
+#include "base/strings/string_number_conversions.h"
 #include "base/system/sys_info.h"
 
 namespace blink {
 
 unsigned NavigatorConcurrentHardware::hardwareConcurrency() const {
+  const auto* cosmium_cmd = base::CommandLine::ForCurrentProcess();
+  if (cosmium_cmd->HasSwitch("cosmium-hardware-concurrency")) {
+    unsigned cosmium_value = 0;
+    if (base::StringToUint(
+            cosmium_cmd->GetSwitchValueASCII("cosmium-hardware-concurrency"),
+            &cosmium_value)) {
+      return cosmium_value;
+    }
+  }
   return static_cast<unsigned>(base::SysInfo::NumberOfProcessors());
 }
 
 }  // namespace blink
